java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -2693,12 +2695,44 @@ public T xadd(
         return getThis();
     }
 
+    /**
+     * Reads entries from the given streams.
+     *
+     * @see <a href="https://redis.io/commands/xread/">redis.io</a> for details.
+     * @param keysAndIds An array of <code>Pair</code>s of keys and entry ids to read from. A <code>
+     *     pair</code> is composed of a stream's key and the id of the entry after which the stream
+     *     will be read.
+     * @return Command Response - A <code>{@literal Map<String, Map<String[][]>>}</code> with stream
```

Comment:
Actually, there is `Object[][]`, since we don't convert the data.
Please update the doc and create a task to add conversion - should be added for some FUNCTION commands too.
```suggestion
     * @return Command Response - A <code>{@literal Map<String, Map<Object[][]>>}</code> with stream
```

Reply:
Node client has similar conversion for commands which return `set`

Reply:
I'm converting in the Java client.  You are correct though, we should convert on the rust side.  
I think I can do that...

Reply:
We can't convert this on rust side, unfortunately.
Rust creates an arbitrary `Object[]` for `Value::Array`, even when this array contains strings only. Rust part doesn't know the expected valye types.
`Object[]` cannot be casted to `String[]` (java I hate you), so we need to traverse the response to convert the data to the more user-friendly format.